src/DGtal/shapes/parametric/Astroid2D.ih

```diff
@@ -80,31 +87,35 @@ inline
 double
 DGtal::Astroid2D<T>::parameter( const RealPoint2D & pp ) const
 {
-  RealPoint2D p( pp );
-  p -= myCenter;
-  double angle;
-
-  if ( p[0] == 0. && p[1] > 0. )
-    angle = M_PI/2.0;
-  else if ( p[0] == 0. && p[1] < 0. )
-    angle = 3.0*M_PI/2.0;
-  else if ( p[1] == 0. && p[0] >= 0. )
-    angle = 0.0;
-  else if ( p[1] == 0. && p[0] < 0. )
-    angle = M_PI;
-  else
+  RealPoint2D p( pp - myCenter );
+  double angle = 0.;
+
+  if ( isAlmostEqual(p[0],0.) )
+  {
+    if  (p[1] > 0. )
+      angle = M_PI_2;
+    else
+      angle = 3. * M_PI_2;
+  }
+  else if ( isAlmostEqual(p[1],0.) )
+  {
+    if ( p[0] > 0. )
+      angle = 0.;
+    else
+      angle = M_PI;
+  }
+  else if ( !(isAlmostEqual(myA,0.) && isAlmostEqual(myB,0.)) )
   {
     if( p[1]/p[0] < 0. )
```

Comment:
is this division safe?

Reply:
Yes, it is in an `else` of a test ensuring that p[0] is almost equal to 0. (see l. 93)

Reply:
Great :-)